packages/Button/src/Button.spec.tsx

```diff
@@ -116,7 +137,12 @@ describe('packages/Button', () => {
 
   test(`renders component inside of a React Element/HTML tag based on as prop, even when "href" is set`, () => {
     const { container } = render(
-      <Button as="div" href="http://mongodb.design">
+      <Button
+        as="div"
+        href="http://mongodb.design"
+        variant={Variant.Default}
+        size={Size.Normal}
```

Comment:
These props should definitely not be required, they're in `Button.defaultProps`.

Reply:
That's the only way they passed ts:check

Reply:
I know! But ts:check would not pass without it.

Reply:
That sounds like it'd be a problem for other people using TS, they shouldn't be forced to include these props. I think we should be able to just convert `<Button>` to a function component, using destructuring defaults instead of defaultProps, actually. There isn't even any state. That should simplify these types. I'll push a commit to your branch, if you don't mind 😁 .

Reply:
Sounds good to me :)

Reply:
Pushed! Not sure what exactly was causing `defaultProps` to stop working suddenly, but a function component makes more sense for `<Button>` regardless of that, so I'm not going to dive too deep down that rabbit hole.